### ISSUE TITLE
What if `p->dtor` is `NULL` in `tgc_free`?

### ISSUE THREAD

**Issue by loggerhead** [orangeduck/tgc#5]
I think the logic of `tgc_free` should changed to:

``` c
void tgc_free(tgc_t *gc, void *ptr) {
    tgc_ptr_t *p  = tgc_get_ptr(gc, ptr);
    if (p) {
        if (p->dtor) {
            p->dtor(ptr);
        }
        free(ptr);
        tgc_rem(gc, ptr);
    }
}
```

**Comment by orangeduck** [comment]
Yes I think you're correct, good catch.

**Comment by orangeduck** [comment]
fixed in 88c13cb3b06f461dc5087d4e532d42dc252b3756
